TypeCobol/Compiler/Directives/TypeCobolOptions.cs

```diff
@@ -33,8 +37,66 @@ public bool UseEuroInformationLegacyReplacingSyntax
             set { _useEuroInformationLegacyReplacingSyntax = value; }
         }
 
+        /// <summary>
+        /// Status code of these option.
+        /// </summary>
+        public ReturnCode OptionStatusCode
+        {
+            get;
+            private set;
+        }
+
 #if EUROINFO_RULES
+        /// <summary>
+        /// The Instance of the Cpy Copy names Map
+        /// </summary>
+        private CopyNameMapFile CpyCopyNamesMap { get; set; }
         private bool _useEuroInformationLegacyReplacingSyntax = true;
+
+        private string _CpyCopyNamesMapFilePath;
+        /// <summary>
+        /// Path to the CpyCopyNames file.
+        /// </summary>
+        public string CpyCopyNamesMapFilePath
+        {
+            get
+            {
+                return _CpyCopyNamesMapFilePath; 
+            }
+            set
+            {
+                if (value == null)
+                {
+                    CpyCopyNamesMap = null;
+                }
+                else if (!value.Equals(_CpyCopyNamesMapFilePath))
+                {                    
+                    CpyCopyNamesMap = GetCpyCopiesFile(value);
+                    _CpyCopyNamesMapFilePath = value;
+                }                
+            }
+        }
+        /// <summary>
+        /// Check if using the current Instance, the Given name corresponds to a CPY copy name. 
+        /// </summary>
+        /// <param name="name">The Copy's name</param>
+        /// <returns>true if the name is CPY Copys name, false otherwise.</returns>
+        public bool HasCpyCopy(string name) => CpyCopyNamesMap?.HasCpyCopy(name) ?? false;
+
+        /// <summary>
+        /// Get the file of CPY COPY names to be used.
+        /// </summary>
+        /// <param name="cpyCopiesFilePath"></param>
+        /// <exception cref="Exception">Any exception if an error occurs.</exception>
+        /// <returns>the CopyNameMapFile instance if one is available, null otherwise</returns>
+        public static CopyNameMapFile GetCpyCopiesFile(string cpyCopiesFilePath)
+        {
+            if (cpyCopiesFilePath != null)
+            {
+                return new CopyNameMapFile(cpyCopiesFilePath);
+            }
+            return null;
+        }
```

Comment:
No check on the relative/rooted nature of the path ? Use of root process location ?